tests/unit/zenodo_api_test.py

```diff
@@ -0,0 +1,27 @@
+"""Test zenodo API client."""
+import pydantic
+import pytest
+
+from pudl_archiver.zenodo.api_client import DatasetSettings
+
+
+def test_dataset_settings():
```

Comment:
non-blocking: These tests seem like they're mostly testing if Pydantic works the way we expect it to - which is fine, but we might not need these? Separately, I think we might want some "test to see if our dataset settings updating logic does the right thing" tests, though those seem a bit more annoying to write.

Reply:
My thoughts were that these sort of test that the regex is working as expected by saying "give it something that looks like a DOI and it's all good. Give it anything else and it fails". Tests that actually test that the settings are handled/updated properly would definitely be more useful though.